### DISCUSSION TITLE
Microsoft ADFS and PowerDNS-Admin

### DISCUSSION THREAD

**Discussion by zotsenko** [ngoduykhanh/PowerDNS-Admin#1492] [Q&A]
Hello.
Im trying to configure Microsoft ADFS and PowerDNS-Admin.
First problem:
When i added SAML integration in config its working, i can login, but im dont have **Sign in with a certificate** button on ADFS.
This is my SAML ADFS config:
# SAML Authnetication
SAML_ENABLED = True
SAML_DEBUG = True
SAML_PATH = os.path.join(os.path.dirname(__file__), 'saml')
SAML_METADATA_URL = 'https://mysite.com/FederationMetadata/2007-06/FederationMetadata.xml'
SAML_METADATA_CACHE_LIFETIME = 1
SAML_IDP_SSO_BINDING = 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST'
SAML_NAMEID_FORMAT = 'urn:oasis:names:tc:SAML:2.0:nameid-format:transient'
SAML_ATTRIBUTE_GIVENNAME = 'firstname'
SAML_ATTRIBUTE_SURNAME = 'lastname'
SAML_SP_ENTITY_ID = 'https://pdnsa.mysite.com'
SAML_SP_CONTACT_NAME = 'me'
SAML_SP_CONTACT_MAIL = 'me'
SAML_CERT = 'saml_cert.crt'
SAML_KEY = 'saml_cert.key'
SAML_SIGN_REQUEST = True
SAML_SIGN_REQUEST = False
SAML_ASSERTION_ENCRYPTED = False
SAML_WANT_MESSAGE_SIGNED = False

Second problem:
When i try configure Microsoft ADFS and OpenID Connect OAuth Settings i have **Sign in with a certificate** button on ADFS but i cant login using login/password or certicicate. 
After entering the username and password on my adfs side I get this error: 

INFO - 127.0.0.1 - - [29/Mar/2023 15:59:00] "GET /login HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/web/python/lib/python3.9/site-packages/flask/app.py", line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File "/web/python/lib/python3.9/site-packages/werkzeug/middleware/proxy_fix.py", line 187, in __call__
    return self.app(environ, start_response)
  File "/web/python/lib/python3.9/site-packages/flask/app.py", line 2076, in wsgi_app
    response = self.handle_exception(e)
  File "/web/python/lib/python3.9/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/web/python/lib/python3.9/site-packages/flask/app.py", line 1519, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/web/python/lib/python3.9/site-packages/flask/app.py", line 1517, in full_dispatch_request
    rv = self.dispatch_request()
  File "/web/python/lib/python3.9/site-packages/flask/app.py", line 1503, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/opt/powerdns-admin/powerdnsadmin/routes/index.py", line 370, in login
    me = json.loads(oidc.get('userinfo').text)
  File "/usr/local/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

Here my config:
![image](https://user-images.githubusercontent.com/62392432/228552019-50a632cb-c173-4a6a-8423-fe284193c64b.png)

**Comment by AzorianMatt** [comment]
@zotsenko based on the discussion you opened, it's really unclear if you're seeking answers to your questions or if you're trying to file a bug report? If the latter is true, that should be done formally through the bug submission process as an issue and not a discussion.

Please clarify or rectify accordingly, thank you.

**Comment by zotsenko** [comment]
@AzorianMatt sorry, i dont know where i should post my question so i did it here. If i did wrong please get me back on track.
I'm just looking for help. My main goal its login to powerdns admin through Microsoft ADFS using certificate. And it doesn't matter if it uses saml or openid. Maybe someone here have experience with configuring scheme like this and can tel where i'm on wrong way.

**Reply by AzorianMatt** [reply]
No worries at all @zotsenko. I just wasn't clear on what you were looking for as a response. Thank you for the follow up.
